fetch-in-react-practice/client/components/app.jsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import PageTitle from './page-title';
+import TodoList from './todo-list';
+import TodoForm from './todo-form';
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: []
+    };
+    this.addTodo = this.addTodo.bind(this);
+    this.toggleCompleted = this.toggleCompleted.bind(this);
+  }
+
+  componentDidMount() {
+    this.getAllTodos();
+  }
+
+  getAllTodos() {
+    fetch('/api/todos')
+      .then(res => res.json())
+      .then(users => this.setState({
+        todos: users
+      }));
+  }
+
+  addTodo(newTodo) {
+    fetch('/api/todos', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(newTodo)
+    })
+      .then(response => response.json())
+      .then(data => {
+        this.setState({ todos: data });
+      })
+      .catch(error => {
+        console.error('Error:', error);
+      });
+  }
+
+  toggleCompleted(todoId) {
+    const todos = this.state.todos;
+    let completedIndex = 0;
+    todos.forEach((obj, index, array) => {
+      if (obj.todoId === todoId) {
+        completedIndex = index;
+      }
+    });
+    const isCompletedStatus = !todos[completedIndex].isCompleted;
+
+    fetch(`/api/todos/${todoId}`, {
+      method: 'PATCH',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({ isCompleted: isCompletedStatus })
+    })
+      .then(response => response.json())
+      .then(data => {
+        const newTodo = this.state.todos;
```

Comment:
Here you want to make a copy of the array first before making changes. Just setting it to another variable does not copy the array. Here is a good article that explains how to copy an array and why it is needed:

https://www.samanthaming.com/tidbits/35-es6-way-to-clone-an-array/